packages/dds/map/src/directory.ts

```diff
@@ -693,17 +698,34 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         return this.localValueMaker.fromSerializable(serializable);
     }
 
+    private getParentDirectory(relativePath: string): IDirectory | undefined {
+        const absolutePath = this.makeAbsolute(relativePath);
+        const subdirs = absolutePath.substr(1).split(posix.sep);
+        if (subdirs.length === 0) {
+            return undefined;
+        }
+        subdirs.pop();
+        return this.getWorkingDirectory(subdirs.join(posix.sep));
+    }
+
+    private isSubDirectoryDeletePending(relativePath: string): boolean {
+        const parentSubDir = this.getParentDirectory(relativePath) as SubDirectory | undefined;
+        const index = relativePath.lastIndexOf(posix.sep);
+        const dirName = relativePath.substring(index + 1);
+        return !!(parentSubDir?.isSubDirectoryDeletePending(dirName));
```

Comment:
Is this recursive? I.e. if any parent is pending delete, will it return true? If not, should it be? As any delete in the chain should be handled the same, right?

Reply:
This is not recursive. This just checks in the immediate parent directory as we don't need or reach here if any parent has been deleted because getWorkingDirectory before where it is called would return empty result.

Reply:
This is not recursive. This just checks in the immediate parent directory as we don't need or reach here if any parent has been deleted because getWorkingDirectory before where it is called would return empty result. However, in cases where this directory has again been created after deletion, then we will be able to reach here. So, I would say,
a.) It is maybe unlikely that if any ancestor is deleted and then entire structure will be created again from deleted parent to this child. 
b.) I think if we need to check entire path from root -> this child, whether node is deleted or not then at that point in time the cons would outweigh the benefit of not applying the operation. So, I think just checking this for immediate parent should be ok.

Reply:
So, if I got you right, you are saying that pending delete for any parent in the chain would result in this function returning false, right? Can you please add a comment on that in the header of the function?
I do not think #a is unlikely - we should handle it properly.
I'm not sure what you mean by cons would outweigh benefits. It all starts with clearly describing behaviors for users. If I delete /A, I expect all objects under it to be gone and any concurrent operations from other machines underneath /A to be dropped on the floor.